### PR TITLE
test: add test for aggregator 

### PR DESCRIPTION
**Pull request by rachit77**
## Description

Aggregator unit tests

### CODE REVIEW

File: aggregator/aggregator_test.go [hunk 1]
```diff
@@ -0,0 +1,1576 @@
...
+	a.halted.Store(false)
+	a.handleRollbackBatches(rollbackData)
+
+	assert.False(t, a.halted.Load())
+	mockStreamClient.AssertExpectations(t)
+	mockEtherman.AssertExpectations(t)
+	mockState.AssertExpectations(t)
+}
+
+func Test_handleReceivedDataStream_BATCH_START(t *testing.T) {
```
Comment (Stefan-Ethernal) [inline aggregator/aggregator_test.go:161]:
```suggestion
func Test_handleReceivedDataStream_BatchStart(t *testing.T) {
```

File: aggregator/aggregator_test.go [hunk 2]
```diff
@@ -0,0 +1,1576 @@
...
+	err = agg.handleReceivedDataStream(batchStartEntry, nil, nil)
+	assert.NoError(t, err)
+
+	assert.Equal(t, agg.currentStreamBatch.BatchNumber, uint64(1))
+	assert.Equal(t, agg.currentStreamBatch.ChainID, uint64(2))
+	assert.Equal(t, agg.currentStreamBatch.ForkID, uint64(3))
+	assert.Equal(t, agg.currentStreamBatch.Type, datastream.BatchType_BATCH_TYPE_REGULAR)
+}
+
+func Test_handleReceivedDataStreamBatchEnd(t *testing.T) {
```
Comment (Stefan-Ethernal) [inline aggregator/aggregator_test.go:198]: Would it be possible to consolidate all the `handleReceivedDataStream` to a single table unit test? Although I'm aware the setup is pretty different, depending which stream data is received.
```suggestion
func Test_handleReceivedDataStream_BatchEnd(t *testing.T) {
```

File: aggregator/aggregator_test.go [hunk 3]
```diff
@@ -0,0 +1,1576 @@
...
+
+	assert.Equal(t, a.currentBatchStreamData, []byte{})
+	assert.Equal(t, a.currentStreamBatchRaw, state.BatchRawV2{Blocks: make([]state.L2BlockRaw, 0)})
+	assert.Equal(t, a.currentStreamL2Block, state.L2BlockRaw{})
+
+	mockState.AssertExpectations(t)
+	mockL1Syncr.AssertExpectations(t)
+}
+
+func Test_handleReceivedDataStreamL2Block(t *testing.T) {
```
Comment (Stefan-Ethernal) [inline aggregator/aggregator_test.go:273]:
```suggestion
func Test_handleReceivedDataStream_L2Block(t *testing.T) {
```

File: aggregator/aggregator_test.go [hunk 4]
```diff
@@ -0,0 +1,1576 @@
...
+	assert.Equal(t, uint64(10), a.currentStreamL2Block.BlockNumber)
+	assert.Equal(t, uint32(5), a.currentStreamL2Block.ChangeL2BlockHeader.DeltaTimestamp)
+	assert.Equal(t, uint32(1), a.currentStreamL2Block.ChangeL2BlockHeader.IndexL1InfoTree)
+	assert.Equal(t, 0, len(a.currentStreamL2Block.Transactions))
+	assert.Equal(t, uint32(1), a.currentStreamBatch.L1InfoTreeIndex)
+	assert.Equal(t, common.BytesToAddress([]byte{0x01}), a.currentStreamBatch.Coinbase)
+	assert.Equal(t, common.BytesToHash([]byte{0x02}), a.currentStreamBatch.GlobalExitRoot)
+}
+
+func Test_handleReceivedDataStreamTransaction(t *testing.T) {
```
Comment (Stefan-Ethernal) [inline aggregator/aggregator_test.go:315]:
```suggestion
func Test_handleReceivedDataStream_Transaction(t *testing.T) {
```

File: aggregator/aggregator_test.go [hunk 5]
```diff
@@ -0,0 +1,1576 @@
...
+			ethTxManager := mocks.NewEthTxManagerClientMock(t)
+			etherman := mocks.NewEthermanMock(t)
+			aggLayerClient := mocks.NewAgglayerClientInterfaceMock(t)
+
+			curve := elliptic.P256()
+			privateKey, err := ecdsa.GenerateKey(curve, rand.Reader)
+			if err != nil {
+				t.Fatal("Error generating key")
+			}
+
```
Comment (Stefan-Ethernal) [inline aggregator/aggregator_test.go:661]:
```suggestion
			require.NoError(t, err, "error generating key")
```

File: aggregator/aggregator_test.go [hunk 6]
```diff
@@ -0,0 +1,1576 @@
...
+			stateMock := mocks.NewStateInterfaceMock(t)
+			ethTxManager := mocks.NewEthTxManagerClientMock(t)
+			etherman := mocks.NewEthermanMock(t)
+			aggLayerClient := mocks.NewAgglayerClientInterfaceMock(t)
+
+			curve := elliptic.P256()
+			privateKey, err := ecdsa.GenerateKey(curve, rand.Reader)
+			if err != nil {
+				t.Fatal("Error generating key")
+			}
```
Comment (Stefan-Ethernal) [inline aggregator/aggregator_test.go:453]:
```suggestion
	require.NoError(t, err, "error generating key")
```

File: aggregator/aggregator_test.go [hunk 7]
```diff
@@ -0,0 +1,1576 @@
...
+		big.NewInt(1000000000000000000),
+		uint64(21000),
+		big.NewInt(20000000000),
+		nil,
+	)
+
+	// Encode transaction into RLP format
+	var buf bytes.Buffer
+	if err := tx.EncodeRLP(&buf); err != nil {
+		t.Fatalf("Failed to encode transaction: %v", err)
```
Comment (Stefan-Ethernal) [inline aggregator/aggregator_test.go:337]: You can use the `require.NoError` here as well.

File: go.mod [hunk 8]
```diff
@@ -3,6 +3,7 @@ module github.com/0xPolygon/cdk
 go 1.22.4
 
 require (
+	github.com/0xPolygon/agglayer v0.1.4
```
[the remark this answers is not in the document]
Comment (ToniRamirezM) [reply go.mod:6]: I would not relay on an archived repo, for sure. In this case I think it makes sense to replicate that code here.

**Comment by ToniRamirezM** [comment]
I think @Stefan-Ethernal did a great review of the PR and I agree with him and his proposals. On the other han I think the changed done in the aggregator itself are ok. I would re-review once Stephan's changes get implemented.

**Review by rbpol — APPROVED** [review]

**Review by ToniRamirezM — APPROVED** [review]

**Pull request merged**